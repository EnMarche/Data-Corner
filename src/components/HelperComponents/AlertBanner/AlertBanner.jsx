/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import Alert from '@material-ui/lab/Alert';
import {
    makeStyles, createStyles,
} from '@material-ui/core';
import PropTypes from 'prop-types';

const useStyles = makeStyles((theme) => createStyles({
    errorMessage: {
        color: theme.palette.statusError,
        background: theme.palette.backgroundError,
        borderRadius: '8.35px',
        elevation: 'none',
    },
}));

function AlertBanner({ severity, message }) {
    const classes = useStyles();

    return (
        <Alert className={classes.errorMessage} elevation={0} variant="filled" severity={severity}>
            {JSON.stringify(message.message)}
        </Alert>
    );
}

export default AlertBanner;

AlertBanner.propTypes = {
    severity: PropTypes.string.isRequired,
    message: PropTypes.string.isRequired,
};
