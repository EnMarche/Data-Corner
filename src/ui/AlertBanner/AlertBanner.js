import Alert from '@mui/material/Alert'
import { makeStyles } from '@mui/styles'
import createStyles from '@mui/styles/createStyles'
import PropTypes from 'prop-types'

const useStyles = makeStyles(theme =>
  createStyles({
    errorMessage: {
      color: theme.palette.statusError,
      background: theme.palette.backgroundError,
      borderRadius: '8.35px',
      elevation: 'none',
    },
  })
)

function AlertBanner({ severity, message }) {
  const classes = useStyles()

  const formatMessage = () => {
    if (message && typeof message === 'object') {
      return (
        <Alert className={classes.errorMessage} elevation={0} variant="filled" severity={severity}>
          {message.message}
        </Alert>
      )
    }
    return (
      <Alert className={classes.errorMessage} elevation={0} variant="filled" severity={severity}>
        {message}
      </Alert>
    )
  }

  return <>{message && formatMessage()}</>
}

export default AlertBanner

AlertBanner.propTypes = {
  severity: PropTypes.string.isRequired,
  message: PropTypes.oneOfType([
    PropTypes.string.isRequired,
    PropTypes.shape({
      message: PropTypes.string.isRequired,
    }),
  ]).isRequired,
}
