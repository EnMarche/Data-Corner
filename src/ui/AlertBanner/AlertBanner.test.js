import { render } from '@testing-library/react'
import AlertBanner from 'ui/AlertBanner/AlertBanner'

jest.mock('@mui/lab/Alert', () => ({ children, ...rest }) => <div {...rest}>{children}</div>)
jest.mock('@mui/material', () => ({
  makeStyles: () => () => ({ errorMessage: 'errorMessage' }),
}))

describe('AlertBanner', () => {
  it('displays AlertBanner with string message', () => {
    const { container } = render(<AlertBanner severity="foo-severity" message="FOO" />)

    expect(container).toMatchSnapshot()
  })
  it('displays AlertBanner with object message', () => {
    const { container } = render(<AlertBanner severity="foo-severity" message={{ message: 'FOO' }} />)

    expect(container).toMatchSnapshot()
  })
})
