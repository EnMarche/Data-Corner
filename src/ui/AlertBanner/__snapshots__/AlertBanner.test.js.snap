// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertBanner displays AlertBanner with object message 1`] = `
<div>
  <div
    class="errorMessage"
    elevation="0"
    severity="foo-severity"
    variant="filled"
  >
    FOO
  </div>
</div>
`;

exports[`AlertBanner displays AlertBanner with string message 1`] = `
<div>
  <div
    class="errorMessage"
    elevation="0"
    severity="foo-severity"
    variant="filled"
  >
    FOO
  </div>
</div>
`;
